jax/_src/numpy/lax_numpy.py

```diff
@@ -3417,17 +3417,22 @@ def _take(a, indices, axis: Optional[int] = None, out=None, mode=None):
 
 def _normalize_index(index, axis_size):
   """Normalizes an index value in the range [-N, N) to the range [0, N)."""
-  if issubdtype(_dtype(index), np.unsignedinteger):
-    return index
-  if core.is_constant_dim(axis_size):
-    axis_size_val = _lax_const(index, axis_size)
-  else:
-    axis_size_val = lax.convert_element_type(core.dimension_as_value(axis_size),
-                                             _dtype(index))
-  return lax.select(
-    lax.lt(index, _lax_const(index, 0)),
-    lax.add(index, axis_size_val),
-    index)
+  try:
+    idx = core.concrete_or_error(operator.index, index, "normalize index")
+    ndim = core.concrete_or_error(operator.index, axis_size, "normalize axis_size")
+    return _canonicalize_axis(idx, ndim)
+  except (TypeError, ValueError, IndexError, core.ConcretizationTypeError):
+    if issubdtype(_dtype(index), np.unsignedinteger):
+      return index
+    if core.is_constant_dim(axis_size):
+      axis_size_val = _lax_const(index, axis_size)
```

Comment:
The rest of this should probably not be in the `except` block, because it will cause confusing tracebacks if something in here raises an error. perhaps use `except: pass` followed by the rest?

Reply:
👍  Changed it. This reduces the diff of the PR.